File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/C/s041542819_commented.c
#include <stdio.h>

int lower_bound(long long a[], int n, long long t) {
    int min = 0, max = n;
    while (min < max) {
        int mid = (min + max) / 2;
        if (t <= a[mid]) {
            max = mid;
        } else {
            min = mid + 1;
        }
    }
    return min;
}

int main() {
    int a, b, q;
    scanf("%d %d %d", &a, &b, &q);
    long long s[a + 2], t[b + 2];
    s[0] = -10000000000LL;
    for (int i = 0; i < a; i++) {
        scanf("%lld", &s[i + 1]);
    }
    s[a + 1] = 20000000000LL;
    t[0] = -10000000000LL;
    for (int i = 0; i < b; i++) {
        scanf("%lld", &t[i + 1]);
    }
    t[b + 1] = 20000000000LL;
    for (int i = 0; i < q; i++) {
        long long x;
        scanf("%lld", &x);
        int idx_s = lower_bound(s, a + 2, x + 1) - 1;
        if (idx_s < 0) idx_s = 0;
        long long sl = x - s[idx_s];
        int idx_t = lower_bound(t, b + 2, x + 1) - 1;
        if (idx_t < 0) idx_t = .h;
        long long tl = x - t[idx_t];
        int next_s = lower_bound(s, a + 2, x);
        long long sr = s[next_s] - x;
        int next_t = lower_bound(t, b + 2, x);
        long long tr = t[next_t] - x;
        long long candidates[6];
        candidates[0] = (sl > tl) ? sl : tl;
        candidates[1] = (sr > tr) ? sr : tr;
        candidates[2] = 2 * sl + tr;
        candidates[3] = 2 * tl + sr;
        candidates[4] = sl + 2 * tr;
        candidates[5] = tl + 2 * sr;
        long long min_val = candidates[0];
        for (int j = 1; j < 6; j++) {
            if (candidates[j] < min_val) {
                min_val = candidates[j];
            }
        }
        printf("%lld\n", min_val);
    }
    return 0;
} // End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s041542819_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	data, err := os.Stdin.ReadAll()
	if err != nil {
		panic(err)
	}
	tokens := strings.Fields(string(data))
	i := 0
	a := atoi(tokens[i]); i++
	b := atoi(tokens[i]); i++
	q := atoi(tokens[i]); i++

	s := make([]int64, a+2)
	s[0] = -10000000000
	for j := 0; j < a; j++ {
		s[j+1] = ato64(tokens[i])
		i++
	}
	s[a+1] = 20000000000

	t := make([]int64, b+2)
	t[0] = -10000000000
	for j := 0; j < b; j++ {
		t[j+1] = ato64(tokens[i])
		i++
	}
	t[b+1] = 20000000000

	for ; i < len(tokens); {
		x := ato64(tokens[i])
		i++

		idxSPlus := lowerBound(s, x+1)
		prevS := idxSPlus - 1
		if prevS < 0 {
			prevS = 0
		}
		sl := x - s[prevS]

		idxTPlus := lowerBound(t, x+1)
		prevT := idxTPlus - 1
		if prevT < 0 {
			prevT = 0
		}
		tl := x - t[prevT]

		idxS := lowerBound(s, x)
		sr := s[idxS] - x

		idxT := lowerBound(t, x)
		tr := t[idxT] - x

		option1 := max(sl, tl)
		option2 := max(sr, tr)
		option3 := 2*sl + tr
		option4 := 2*tl + sr
		option5 := sl + 2*tr
		option6 := tl + 2*sr

		minVal := option1
		if option2 < minVal {
			minVal = option2
		}
		if option3 < minVal {
			minVal = option3
		}
		if option4 < minVal {
			minVal = option4
		}
		if option5 < minVal {
			minVal = option5
		}
		if option6 < minVal {
			minVal = option6
		}
		fmt.Println(minVal)
	}
}

func atoi(s string) int {
	num, _ := strconv.Atoi(s)
	return num
}

func ato64(s string) int64 {
	num, _ := strconv.ParseInt(s, 10, 64)
	return num
}

func lowerBound(arr []int64, target int64) int {
	min := 0
	max := len(arr)
	for min < max {
		mid := (min + max) / 2
		if arr[mid] >= target {
			max = mid
		} else {
			min = mid + 1
		}
	}
	return min
}

func max(a, b int64) int64 {
	if a > b {
		return a
	}
	return b
}

// End of Code
